File: python/Exhaustive_Search/primeNumber.py
from itertools import permutations
def solution(numbers):
    numbers = list(numbers)
    check = set()
    for i in range(1, len(numbers)+1):
        base = list(permutations(numbers, i))
        for j in base:
            tmp = ''.join(map(str, j))
            if int(tmp) == 2:
                check.add(int(tmp))
                continue
            elif int(tmp) in [0,1]: continue
            l = int(int(tmp) ** 0.5)
            for k in range(2,l+1):
                if not int(tmp)%k:
                    break
            else:
                check.add(int(tmp))
    answer = len(check)
    return answer

print(solution('17'))
print(solution('011'))

File: javascript/Exhaustive_Search/primeNumber.js
function solution(numbers) {
    var answer = 0;
    
    let n = numbers.split('');
    let nums = new Set();
    
    function combi(a, s) {
        if (s.length > 0) {
            if (nums.has(Number(s))===false) {
                nums.add(Number(s));
                
                if (checkPrime(Number(s))) {
                    answer++;
                }
            }
        }
        
        if (a.length > 0) {
            for (let i=0; i < a.length; i++) {
                let t = a.slice(0);
                t.splice(i,1);
                combi(t, s+a[i])
            }
        }
    }
    
    function checkPrime(num) {
        if (num<2) return false;
        if (num===2) return true;
        for (let i=2; i <= Math.sqrt(num); i++) {
            if (num%i===0) return false
        }
        return true
    }
    combi(n, '');
    return answer
}

console.log(solution("17"))
console.log(solution("011"))